src/main/java/org/veupathdb/service/eda/ds/plugin/AbstractPlugin.java

```diff
@@ -610,4 +641,138 @@ public String getRBinListAsString(List<String> labels) {
     return rBinList + "))";
   }
 
+  public boolean validateImputeZeroesRequest(Map<String, DynamicDataSpecImpl> dataSpecs) {
+    // TODO check a bunch of stuff i no longer remember
+    return true;
+  }
+
+  public String getRStudyVocabsAsString(DynamicDataSpecImpl dataSpec, ResponseFuture studyVocab) {
+    PluginUtil util = getUtil();
+    
+    // this assuming the first ancestor is the root one is a bit hacky, but i did the same in R so...
+    // obviously doing the same awful thing twice makes it ok. but doing any better is higher cost than i have time for right now :(
+    String readStudyVocabInR = "data.table::fread(" + studyVocab + ", header = FALSE)";
```

Comment:
I suspect this isn't going to work as is. `studyVocab` is just going to return a string with the memory address of the `ResponseFuture`. I suspect to make this available as a file in R you'll need to use the RFileSetProcessor in some way or you could just parse the whole thing to a string representation of the R table in Java.

I'll look more tomorrow morning at how the former recommendation would work. I'm not that familiar as of now with our Java -> R communication.